### PR TITLE
PR: add url for favicon middleware, for correct handling different of…

### PR DESCRIPTION
**Pull request by 0xdeface**
## Description
Presently ico url might be only /favicon.ico but some resource wants url exactly /favicon.svg for svg format.
I added new property to favicon config and default value with /favicon.ico. Also i changed docs for this middleware

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [x] I tried to make my code as fast as possible with as few allocations as possible
- [x] For new code I have written benchmarks so that they can be analyzed and improved

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by li-jin-gou** [comment]
Add pr Description.

File: middleware/favicon/favicon.go [hunk 1]
```diff
@@ -20,7 +20,10 @@ type Config struct {
 	//
 	// Optional. Default: ""
 	File string `json:"file"`
-
+	// Url for favicon handler
+	//
+	// Optional. Default: "/favicon.ico
+	Url string `json:"url"`
```
Comment (efectn) [inline middleware/favicon/favicon.go:26]: URL would be better naming i think

**Review by efectn — APPROVED** [review]

File: middleware/favicon/favicon.go [hunk 2]
```diff
@@ -99,7 +106,7 @@ func New(config ...Config) fiber.Handler {
 		}
 
 		// Only respond to favicon requests
-		if c.Path() != fPath {
+		if c.Path() != cfg.URL {
```
Comment (efectn) [inline middleware/favicon/favicon.go:109]: Can you add a testcase

**Comment by pjebs** [comment]
Shouldn't the file be cached using sync.Once?

**Comment by 0xdeface** [comment]
Sorry, I did not understand,  what I should do? Use sync.once?

**Comment by 0xdeface** [comment]
> Shouldn't the file be cached using sync.Once?

i didn't change any cache behaviour in this pr.  please explain where i should add sync.once?

**Comment by ReneWerner87** [comment]
I'll try to take a look at the pull request tomorrow, if I know what he means or have other comments I'll let you know

**Comment by pjebs** [comment]
My mistake. I thought the middleware's job was to return a (cached) binary version of the favicon.

**Comment by ReneWerner87** [comment]
@0xdeface could you not save the internal configuration switch by specifying the route when using the middleware ?

example
```go
app.Use("/favicon.svg", favicon.New())
```
you could make this feature much more usable and would only have to deactivate the internal matching in such a case.

File: middleware/favicon/README.md [hunk 3]
```diff
@@ -67,6 +72,7 @@ type Config struct {
 ```go
 var ConfigDefault = Config{
 	Next: nil,
-	File:	""
+	File:	"",
+	URL: "/favicon.ico"
```
Comment (leonklingele) [inline middleware/favicon/README.md:76]: This wouldn't compile without a slash at the end of line here.

File: middleware/favicon/README.md [hunk 4]
```diff
@@ -54,11 +55,15 @@ type Config struct {
...
-
+    
 	// File holds the path to an actual favicon that will be cached
 	//
 	// Optional. Default: ""
 	File string
+	// URL for favicon handler
+	//
+	// Optional. Default: "/favicon.ico
+	URL string `json:"url"`
```
Comment (leonklingele) [inline middleware/favicon/README.md:66]: Why do we need the `json` annotation here?

File: middleware/favicon/README.md [hunk 5]
```diff
@@ -42,6 +42,7 @@ app.Use(favicon.New())
 ```go
 app.Use(favicon.New(favicon.Config{
 	File: "./favicon.ico",
+	URL: "/favicon.ico"
```
Comment (leonklingele) [inline middleware/favicon/README.md:45]: Please format, i.e. add another space character between the colon and the opening quote.

File: middleware/favicon/README.md [hunk 6]
```diff
@@ -54,11 +55,15 @@ type Config struct {
 	//
 	// Optional. Default: nil
 	Next func(c *fiber.Ctx) bool
-
+    
 	// File holds the path to an actual favicon that will be cached
 	//
 	// Optional. Default: ""
 	File string
+	// URL for favicon handler
```
Comment (leonklingele) [inline middleware/favicon/README.md:63]: Please add a blank line before this line.

File: middleware/favicon/README.md (hunk 3 above)
Comment (leonklingele) [reply middleware/favicon/README.md:76]: Also, please properly align. Also the `File` line seems to include a redundant space.

File: middleware/favicon/favicon.go [hunk 7]
```diff
@@ -20,7 +20,10 @@ type Config struct {
 	//
 	// Optional. Default: ""
 	File string `json:"file"`
-
+	// URL for favicon handler
+	//
+	// Optional. Default: "/favicon.ico
+	URL string `json:"url"`
```
Comment (leonklingele) [inline middleware/favicon/favicon.go:26]: See above, why the `json` annotation?

File: middleware/favicon/README.md [hunk 8]
```diff
@@ -54,11 +55,15 @@ type Config struct {
...
+	//
+	// Optional. Default: "/favicon.ico
```
Comment (leonklingele) [inline middleware/favicon/README.md:65]: Missing closing quotes.

File: middleware/favicon/favicon.go [hunk 9]
```diff
@@ -20,7 +20,10 @@ type Config struct {
 	//
 	// Optional. Default: ""
 	File string `json:"file"`
-
+	// URL for favicon handler
+	//
+	// Optional. Default: "/favicon.ico
```
Comment (leonklingele) [inline middleware/favicon/favicon.go:25]: Missing closing quotes.

**Comment by leonklingele** [comment]
Also make sure to update `middleware/favicon/README.md` accordingly. Currently it says `This middleware is exclusively for serving the default, implicit favicon, which is GET /favicon.ico`.

**Comment by ReneWerner87** [comment]
@0xdeface Hi can check the last comments and suggestions for improvement

hope we can still customize it so that we provide this feature in the next version

**Comment by 0xdeface** [comment]
Hi,  i will check comments and put changes on next week.

**Comment by leonklingele** [comment]
@0xdeface did you get a chance to work on this so far? 😊

File: middleware/favicon/README.md [hunk 10]
```diff
@@ -54,11 +55,15 @@ type Config struct {
 	//
 	// Optional. Default: nil
 	Next func(c *fiber.Ctx) bool
-
+
```
Comment (renanbastos93) [inline middleware/favicon/README.md:58]:
```suggestion

```

**Review by renanbastos93 — COMMENTED** [review]
LGTM, please consider suggestion by @leonklingele

**Comment by efectn** [comment]
@leonklingele can you check the PR again

**Review by renanbastos93 — APPROVED** [review]

**Review by leonklingele — APPROVED** [review]
lgtm, good work @0xdeface! :partying_face:

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**